drivers/netdev2_ieee802154/netdev2_ieee802154.c

```diff
@@ -170,6 +170,16 @@ int netdev2_ieee802154_set(netdev2_ieee802154_t *dev, netopt_t opt, void *value,
     int res = -ENOTSUP;
 
     switch (opt) {
+        case NETOPT_CHANNEL:
+            /* validity needs to be checked by device, since sub-GHz and 2.4 GHz
+             * band radios have different legal values */
+            if (len > sizeof(dev->chan)) {
+                res = -EOVERFLOW;
+                break;
+            }
+            dev->chan = *((uint8_t *)value);
```

Comment:
is chan `uint8_t` here? It is defined as `uint16_t` in `netopt.h` -> we should be consistent here, though I would not arguing against changing this in `netopt.h`...